Katello/katello
# Refs #34191 - New host details - Install Packages modal

- [x] add katello-agent support

#### What are the changes introduced in this pull request?
![package_install_modal](https://user-images.githubusercontent.com/22042343/149962531-686ae1e8-61de-4ae6-9e99-bbc04b838f68.png)

* Adds 'Install packages' action to kebab
* Action brings up modal as per mockup
* You can select packages and choose how to install them


![package_install_action_menu](https://user-images.githubusercontent.com/22042343/149962529-7e9661a3-76e1-4223-acfd-4942b32d4b51.png)

#### Considerations taken when implementing this change?

* Existing remote execution feature `katello_package_install` was left alone
* New REX feature `katello_package_install_by_search_query` added - this accepts package IDs as part of the search query so it works with our existing SelectAll infrastructure

#### What are the testing steps for this pull request?

To import the new job template:
```rb
# in Rails console
ForemanInternal.all.first.delete
```
Then run
```
bundle exec rails db:seed
```
In Hosts > Job Templates, you'll see a new locked template:
```
Install packages by search query - Katello SSH Default
```

Now you're ready to test.

1. Register a rhel7 host and sync some Red Hat repositories
2. Visit the new host details page > Content > Packages
3. Action kebab > Install packages
4. Test both 'Install via remote execution' and 'Install via customized remote execution'

things to test:
- make sure the list of available packages looks sane - test this with both a library and non-library host
- open the modal, close the modal, open it again - The modal reappears as before, without making a second API request and reloading the data
- When no items are selected, the 'Install' button and dropdown are disabled - it becomes enabled when you select an item
- Click on the 'Install' button - immediately executes 'Install via remote execution'
- Click the dropdown and then click 'Install via customized remote execution' - you are taken to the job invocation page with IDs filled out as the package search query
- A package search query such as `id ^ (32372, 33474)` should result in the correct command in the rendered template: `yum -y install duck armadillo` - test this both with regular and customized REX
